src/components/program-progress/data/utils.jsx

```diff
@@ -31,3 +32,67 @@ export function getProgramCertImage(type) {
       return '#';
   }
 }
+
+export function getLastEndingCourseDate(courses) {
+  let courseRuns = [];
+  courseRuns = courses.map((course) => (
+    course?.courseRuns.map((cRun) => (cRun
+    ))
+  ));
+  // combine all the course runs in the program which are not started by learner.
+  const ConcatenatedCourseRuns = [].concat(...courseRuns);
+  return ConcatenatedCourseRuns?.sort(
+    (a, b) => (moment(a.end) < moment(b.end) ? 1 : -1),
+  )[0].end;
+}
+
+export function getEnrolledCourseRunDetails(courses) {
+  const enrolledCourseRunDetails = [];
+  courses.map((course) => (
+    course?.courseRuns.map((cRun) => (cRun.isEnrolled && (enrolledCourseRunDetails.push({
+      start: cRun.start,
+      title: cRun.title,
+      key: course?.key,
+      pacingType: cRun?.pacingType,
+      certificateUrl: cRun?.certificateUrl,
+    }))
+    ))
+  ));
+  return enrolledCourseRunDetails;
+}
+
+export function getNotStartedCourseDetails(courses) {
+  const courseWithSingleCourseRun = [];
+  const courseWithMultipleCourseRun = [];
+  const courseAllCourseRuns = [];
+
+  courses.map((course) => (
+    course?.courseRuns?.length && course?.courseRuns.length === 1 ? (courseWithSingleCourseRun.push({
```

Comment:
nit: save this `course?.courseRuns?.length` in a `const` that can be reused elsewhere, this could make code more readable